src/PrestaShopBundle/Localization/Locale.php

```diff
@@ -0,0 +1,264 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Localization;
+
+use InvalidArgumentException;
+use PrestaShop\Decimal\Operation\Rounding;
+use PrestaShopBundle\Currency\CurrencyCollection;
+use PrestaShopBundle\Localization\CLDR\LocaleData;
+use PrestaShopBundle\Localization\CLDR\NumberSymbolList;
+use PrestaShopBundle\Localization\Formatter\Number as NumberFormatter;
+use PrestaShopBundle\Localization\Formatter\NumberFactory as NumberFormatterFactory;
+use PrestaShopBundle\Localization\Repository as LocaleRepository;
+
+class Locale
+{
+    protected $localeCode;
+    protected $repository;
+    protected $numberFormatter;
+    protected $numberFormatterFactory;
+    protected $specification;
+    protected $id;
+    protected $currencyCollection;
+    protected $roundMode = 2; // Round half up (recommended)
```

Comment:
will never be used because overriden during instanciation...

Reply:
Will fix.